### PR TITLE
feat: Add PWA support

### PR DESCRIPTION
**Pull request by nicky-dev**
PWA support
- Install `next-pwa`
- Use next-pwa in next.config.js
- Add manifest.json
- Add meta viewport in pages/_app
- Add manifest attributes in pages/_document

Remember the pubkey
- Read the pubkey from contactListAtom

Issue: #109

### CODE REVIEW

**Comment by rolznz** [comment]
Thanks @nicky-dev for working on this!

I built your branch and ran it locally using ngrok to connect from my phone, then share -> save to homescreen.

One issue:
- on iOS, saving the app to my home screen does not use the provided icon.

File: src/components/nostr/Login.tsx [hunk 1]
```diff
@@ -402,11 +402,18 @@ function LoggedInUser({ pubkey, onClose }) {
 
 export default function Login() {
   const toast = useToast();
+  const [contacts] = useAtom(contactListAtom);
   const [pubkey, setPubkey] = useAtom(pubkeyAtom);
   const [privkey, setPrivkey] = useAtom(privkeyAtom);
   const { isOpen, onOpen, onClose } = useDisclosure();
   const { t } = useTranslation("common");
 
+  useEffect(() => {
```
Comment (rolznz) [inline src/components/nostr/Login.tsx:411]: Is this needed? you can create an account with habla. CC @verbiricha

File: src/pages/_app.tsx [hunk 2]
```diff
@@ -65,13 +66,21 @@ function MyApp({ Component, pageProps }: AppProps) {
...
-        <QueryClientProvider client={queryClient}>
-          <Component {...pageProps} />
-        </QueryClientProvider>
-      </ChakraProvider>
-    </NostrContext.Provider>
+    <>
+      <Head>
+        <meta
+          name="viewport"
+          content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=0"
```
Comment (rolznz) [inline src/pages/_app.tsx:73]: `user-scalable=0` is not considered good practice. Is it needed?

Comment (nicky-dev) [reply src/pages/_app.tsx:73]: Just to prevent unwanted zooming. we can remove it.

**Comment by nicky-dev** [comment]
> Thanks @nicky-dev for working on this!
> 
> I built your branch and ran it locally using ngrok to connect from my phone, then share -> save to homescreen.
> 
> One issue:
> 
> * on iOS, saving the app to my home screen does not use the provided icon.

Thanks @rolznz. 
I tried with ngrok, and on iOS, it can be 'Add to Home Screen' and display the icon.

IMG_2109

**Comment by rolznz** [comment]
> > Thanks @nicky-dev for working on this!
> > I built your branch and ran it locally using ngrok to connect from my phone, then share -> save to homescreen.
> > One issue:
> > 
> > * on iOS, saving the app to my home screen does not use the provided icon.
> 
> Thanks @rolznz. I tried with ngrok, and on iOS, it can be 'Add to Home Screen' and display the icon.
> 
> IMG_2109

Hmm, thanks! it worked this time. :pray:

File: src/components/nostr/Login.tsx (hunk 1 above)
Comment (nicky-dev) [reply src/components/nostr/Login.tsx:411]: Thanks, @rolznz, for helping with the code review.

I added this code so users don't have to log in every time they access habla.new.

Comment (rolznz) [reply src/components/nostr/Login.tsx:411]: This is for read-only access only, right? sounds good!

File: src/pages/_app.tsx (hunk 2 above)
Comment (verbiricha) [reply src/pages/_app.tsx:73]: Perhaps better to remove, don't want to remove ability to zoom in.

**Comment by verbiricha** [comment]
Minor comments, thanks for the PR! There are some conflicts in pnpm-lock after merging #114

**Comment by nicky-dev** [comment]
> Minor comments, thanks for the PR! There are some conflicts in pnpm-lock after merging #114

I've resolved the conflicts.

File: src/components/nostr/Login.tsx (hunk 1 above)
Comment (nicky-dev) [reply src/components/nostr/Login.tsx]: Sure, I got it. I've done it.

File: src/pages/_app.tsx (hunk 2 above)
Comment (nicky-dev) [reply src/pages/_app.tsx]: Alright, I'll revert it.

**Comment by rolznz** [comment]
@nicky-dev thanks for working on this. Bounty paid

image

**Comment by nicky-dev** [comment]
Thanks so much @rolznz, but this PR hasn't been merged yet.

**Comment by rolznz** [comment]
> Thanks @rolznz, but this PR hasn't been merged yet.

It's ok. I think from your side the work has been completed

**Comment by verbiricha** [comment]
Thanks for the work @nicky-dev and @rolznz for the review!

**Pull request merged**